### ISSUE TITLE
Bug: Incorrect type inference for `watchImmediate` with multiple watch sources

### ISSUE THREAD

**Issue by abaza738** [vueuse/vueuse#4153]
### Describe the bug

Callback function for `watchImmediate` with multiple watch sources infers wrong types for parameters.  
The resulting faulty type is a union of all the watch sources types.

![image](https://github.com/user-attachments/assets/bc015157-da9a-4fe9-944d-4df28cc42cc0)

Compared to a normal `watch` with `{ immediate: true }`, the types are inferred correctly.

![image](https://github.com/user-attachments/assets/a2cddc4e-932a-40a1-9660-8b3b460ac273)

### Reproduction

https://github.com/abaza738/vueuse-minimal

### System Info

```Shell
System:
    OS: Windows 11 10.0.22631
    CPU: (12) x64 AMD Ryzen 5 5600 6-Core Processor
    Memory: 20.25 GB / 31.93 GB
  Binaries:
    Node: 20.16.0 - C:\Program Files\nodejs\node.EXE
    npm: 10.8.1 - C:\Program Files\nodejs\npm.CMD
    pnpm: 8.10.5 - ~\AppData\Local\pnpm\pnpm.EXE
  Browsers:
    Edge: Chromium (127.0.2651.74)
    Internet Explorer: 11.0.22621.3527
  npmPackages:
    @vueuse/core: ^10.11.1 => 10.11.1 
    @vueuse/nuxt: ^10.11.1 => 10.11.1 
    vue: latest => 3.4.38
```

### Used Package Manager

pnpm

### Validations

- [X] Follow our [Code of Conduct](https://github.com/vueuse/vueuse/blob/main/CODE_OF_CONDUCT.md)
- [X] Read the [Contributing Guidelines](https://github.com/vueuse/vueuse/blob/main/CONTRIBUTING.md).
- [X] Read the [docs](https://vueuse.org/guide).
- [X] Check that there isn't [already an issue](https://github.com/vueuse/vueuse/issues) that reports the same bug to avoid creating a duplicate.
- [X] Make sure this is a VueUse issue and not a framework-specific issue. For example, if it's a Vue SFC related bug, it should likely be reported to https://github.com/vuejs/core instead.
- [X] Check that this is a concrete bug. For Q&A open a [GitHub Discussion](https://github.com/vueuse/vueuse/discussions).
- [X] The provided reproduction is a [minimal reproducible example](https://stackoverflow.com/help/minimal-reproducible-example) of the bug.

**Comment by OsamaHaikal** [comment]
Updating to version 11.0.0-beta.3 resolves the type inference issue but not sure what was the issue before
@abaza738

**Comment by abaza738** [comment]
Fixed in #4043 with v11.0.0 release.
